lib/spack/llnl/util/tty/log.py

```diff
@@ -430,7 +430,12 @@ def __enter__(self):
             except BaseException:
                 input_stream = None  # just don't forward input if this fails
 
-            self.process = multiprocessing.Process(
+            if sys.version_info >= (3,):  # novm
+                process_factory = multiprocessing.get_context('fork')
+            else:
+                process_factory = multiprocessing
+
+            self.process = process_factory.Process(
```

Comment:
Wrap this logic up in a class, like `llnl.util.lang.ForkProcess`, and use that instead of duplicating code everywhere:

In `llnl.util.lang`:

```python
if sys.version_info >= (3,):
    ForkProcess = multiprocessing.get_context('fork')
else:
    ForkProcess = multiprocessing.Process
```

Then construct an `llnl.util.lang.ForkProcess` in the code above.